### PR TITLE
zcs-2290: add more info on configuring SSDB and performance

### PR DESCRIPTION
**Pull request by grishick**
* Created 2 wikis on github to describe current test results and linked the guide to the wikis.
* added some more information on installation and minimum requirements for SSDB

### CODE REVIEW

**Comment by msmcgillis** [comment]
In both cases I feel like we should find a way to provide the community with the actual jmeter tests so anyone could take the tests we run and reproduce our results. This would probably be a entirely new WIKI and repo but it could be part of a admin guide I suppose. Also has some overhead of how exactly to work this out.

On the LDAP test I wonder if we should provide information on how people can detect that they may be experiencing the problem. Actually spell out how to look at the ldap free table usage. Give them information on how to look at the number of authTokens and CSRFtokens in LDAP for their users. If users understand how to evaluate this and look at it they may find ways to potential track down their own problems or manage things so LDAP doesn't get into bad states. If i'm correct that some of this is actually abuse giving users the knowledge of how to find it may help them to find and shutdown bad accounts. I also think pointing users to how to check replication status would be good the easy way would be referencing libexec/zmreplchk however I wouldn't reference it unless someone fixes it as far as I can tell it is broken. Not sure I would go this far but one process I worked out for LDAP to deal with the freelist was "Clear LDAP Freelist" once your LDAP freelist gets bad from what I can tell doing something like my "Clear LDAP Freelist" process is necessary to get LDAP stable again. But this is only useful after you have cleaned up accounts with large numbers of authTokens and CSRFTokens.

On the SSDB side of things, I have written up in my ZMS testing page things that I had to work out for this testing effort:

SSDB Build
SSDB Install
SSDB HAProxy Build
SSDB HAProxy Install
SSDB Backup
SSDB Restore
SSDB Remove Replication
Zimbra Enable SSDB
Zimbra Disable SSDB

Not sure how much if any of the above belong in any of this the zimbra items probably should have a home some place but the rest I'm not sure zimbra wants or should maintain. Your linking users to SSDB is probably good at least until you decide to bundle SSDB with zimbra.

Outside of the above for the testing purposes my ephemeral script that lets me see tokens in ssdb by user is useful. I also have a script that parses a ssdb log file which again for testing purposes is very useful.

The main thing I would probably highlight on the SSDB side is that so far in our testing is that a very small default SSDB install will easily support a small (wish I had real production data to clarify small but I don't so I can't) zimbra install. Meaning you will experience problems in other zimbra infrastructure before you will ever see a problem with a default SSDB server as used in our test.

File: ephemeraldata.adoc [hunk 1]
```diff
@@ -26,11 +26,15 @@ The two currently supported Ephemeral Data backends are:
 
 Mail systems with large numbers of active users have been found to overload LDAP as noted in  https://bugzilla.zimbra.com/show_bug.cgi?id=104858[Bugzilla 104858].
 
-== Post {product-name} Install Configuration
+Frequent authentication requests place a high load on Ephemeral Data storage backend. See the following Zimbra wiki pages for results of authentication-based laod tests:
+
+* https://github.com/Zimbra/zm-ssdb-ephemeral-store/wiki/Zimbra-and-LDAP-Authentication-Load-Tests[LDAP Authentication load tests]
+* https://github.com/Zimbra/zm-ssdb-ephemeral-store/wiki/Zimbra-and-SSDB-Authentication-Load-Tests[SSDB Authentication load tests]
```
Comment (fatkudu) [inline ephemeraldata.adoc]: The Wiki page could do with mentioning which version of SSDB was under test.

File: ssdb-configuration-options.adoc [hunk 2]
```diff
@@ -1,4 +1,10 @@
-= SSDB Configuration Options
+= SSDB Installation and Configuration
+
+== Installation
+
+{product-name} packages do not include SSDB server and {product-name} installation and configuration utilities do not alter SSDB configuration. To install SSDB, follow instructions provided by SSDB developer community in http://ssdb.io/docs/install.html[SSDB Installation Documentation].
```
Comment (fatkudu) [inline ssdb-configuration-options.adoc]: Those instructions are for getting the latest SSDB.  Fine as far as it goes but I think we should be clear on which version we tested with and perhaps provide an alternative installation method to get the specific version we tested - e.g. based on the script (old now perhaps 1.9.3 is no longer appropriate):
```
#!/bin/bash
SSDB_INSTALL_DIR="/home/zimbra/ssdb"
if [ -d "${SSDB_INSTALL_DIR}" ]
then
    echo "### Assuming ssdb is already setup under ${SSDB_INSTALL_DIR}"
else
    export SSDBVERSION=1.9.3
    export SSDBDIR="${SSDB_INSTALL_DIR}"
    if [ -d ${SSDBDIR} ]
    then
        echo "${SSDBDIR} already exists!"
        exit 1
    fi
    (
        set -x
        sudo apt-get install autoconf
        mkdir -p ${SSDBDIR} &&
        cd ${SSDBDIR} &&
        git clone -n --depth=1 -b stable-${SSDBVERSION} -o origin https://github.com/ideawu/ssdb.git &&
        cd ssdb &&
        git fetch --all &&
        git checkout stable-${SSDBVERSION} &&
        make PREFIX=${SSDBDIR} && make install PREFIX=${SSDBDIR} &&
        ${SSDBDIR}/ssdb-server -d ${SSDBDIR}/ssdb.conf
        ps -ef | grep ssdb-server
        ${SSDBDIR}/ssdb-server -d ${SSDBDIR}/ssdb.conf -s stop
    )
fi
```

File: ssdb-configuration-options.adoc [hunk 3]
```diff
@@ -572,6 +576,11 @@ blocked indefinitely, even after the connection is re-established.
 In general, the resource pool should be sized such that the mailbox server
 is not starved for resources.
 
+== Scaling SSDB for Production Load with {product-name}
+The main characteristics of {product-name} production load that affects load on SSDB server are the frequency of authentication requests and frequency of SOAP requests sent by {product-name} Web Client and 3rd party SOAP clients. Each authentication request results in a 2 or 3 write operations for SSDB. The write operations update zimbraLastLogonTimestamp, zimbraAuthTokens and zimbraCsrfTokenData values . Note, that zimbraCsrfTokenData is updated only when using a CSRF-enabled SOAP client such as {product-name} Web Client. Each authenticated SOAP request results in 2 read operations for SSDB.
```
Comment (plobbes) [inline ssdb-configuration-options.adoc]: nit: remove space before period at end of sentence "...values . Note, ..."

File: ephemeraldata.adoc (hunk 1 above)
Comment (grishick) [reply ephemeraldata.adoc]: done. Added SSDB version to the wiki with test results

**Pull request merged**